src/main/java/org/prebid/server/settings/S3ApplicationSettings.java

```diff
@@ -0,0 +1,224 @@
+        this.storedResponsesDirectory = Objects.requireNonNull(storedResponsesDirectory);
+        this.jacksonMapper = Objects.requireNonNull(jacksonMapper);
+        this.vertx = Objects.requireNonNull(vertx);
+    }
+
+    @Override
+    public Future<Account> getAccountById(String accountId, Timeout timeout) {
+        return downloadFile(accountsDirectory + "/" + accountId + JSON_SUFFIX)
+                .map(fileContentOpt ->
+                        fileContentOpt.map(fileContent -> jacksonMapper.decodeValue(fileContent, Account.class)))
+                .compose(accountOpt -> {
+                    if (accountOpt.isPresent()) {
+                        return Future.succeededFuture(accountOpt.get());
+                    } else {
+                        return Future
+                                .failedFuture(new PreBidException("Account with id %s not found".formatted(accountId)));
+                    }
+                })
+                .flatMap(account -> {
+                    if (!Objects.equals(account.getId(), accountId)) {
+                        return Future.failedFuture(new PreBidException(
+                                "Account with id %s does not match id %s in file".formatted(accountId, account.getId()))
+                        );
+                    }
+                    return Future.succeededFuture(account);
+                })
+                .recover(ex -> {
+                    if (ex instanceof DecodeException) {
+                        return Future
+                                .failedFuture(
+                                        new PreBidException(
+                                                "Invalid json for account with id %s".formatted(accountId)));
+                    }
+                    // if a previous validation already yielded a PreBidException, just return it
+                    if (ex instanceof PreBidException) {
+                        return Future.failedFuture(ex);
+                    }
+                    return Future
+                            .failedFuture(new PreBidException("Account with id %s not found".formatted(accountId)));
+                });
+    }
+
+    @Override
+    public Future<StoredDataResult> getStoredData(
+            String accountId,
+            Set<String> requestIds,
+            Set<String> impIds,
+            Timeout timeout) {
+
+        return getFileContents(storedRequestsDirectory, requestIds)
+                .compose(storedIdToRequest -> getFileContents(storedImpressionsDirectory, impIds)
+                        .map(storedIdToImp ->
+                                buildStoredDataResult(storedIdToRequest, storedIdToImp, requestIds, impIds))
+                );
+    }
+
+    private StoredDataResult buildStoredDataResult(
+            Map<String, String> storedIdToRequest,
+            Map<String, String> storedIdToImp,
+            Set<String> requestIds,
+            Set<String> impIds
+    ) {
```

Comment:
Please, add empty line as I mentioned before :). 
And, please,  place `) {` on previous line (). We added this rule for expressions, but its meant for any enclosing braces. https://github.com/prebid/prebid-server-java/blob/master/docs/developers/code-style.md#parenthesis-placement.

Also, you may wonder - why do I even bother you with such small and seemlessly useless stuff? Its all about consistency. Inconsistent code hard to maintain :(

Reply:
Done